lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -144,7 +145,11 @@ def get_required_content(usage_info, block_structure):
 
         """
         course_key = block_structure.root_block_usage_key.course_key
-        user_can_skip_entrance_exam = EntranceExamConfiguration.user_can_skip_entrance_exam(usage_info.user, course_key)
+        user_can_skip_entrance_exam = False
+        is_enrolled = CourseEnrollment.is_enrolled(usage_info.user, course_key)
+        if is_enrolled:
+            user_can_skip_entrance_exam = EntranceExamConfiguration.user_can_skip_entrance_exam(
+                usage_info.user, course_key)
```

Comment:
Yes, isn't that what is happening? i.e. the initial value `False` is only be updated if the user is enrolled.

Reply:
Gah, sorry @symbolist , you're correct.  I didn't read the full context.

Reply:
Nit/Question: Why isn't the extra logic of "are they enrolled?" pushed into `EntranceExamConfiguration.user_can_skip_entrance_exam`?

Reply:
That is possible; though it may add a database query to `user_can_skip_entrance_exam` if the `CourseEnrollment` is not being called for the user otherwise in that request?

The block structure transformers do need some more tweaks. Currently [`usage_info`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_blocks/usage_info.py) assumes that the user passed to it is going to be authenticated and so transformers which use it throw errors in a few places if an unauthenticated one is passed. This is the only one which runs in a default course setup so needed this change. The next PR is going to look at the transformers along with the cohorts updates so can look at this together.